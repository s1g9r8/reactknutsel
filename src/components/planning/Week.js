
import React from 'react';
import moment from 'moment';

import Day from './Day.js';

const Week = (props) => {

  var { week } = props;

  var createDate = (day, weekNumber, year) => {
    return (moment().day(day).week(weekNumber).year(year).format('DD-MM'));
  };

  return (
    <div>
      <table>
        <caption>Week {week.weekNumber}</caption>
        <thead>
          <th></th>
          <th>Monday {createDate("Monday", week.weekNumber, week.year)}</th>
          <th>Tuesday {createDate("Tuesday", week.weekNumber, week.year)}</th>
          <th>Wednesday {createDate("Wednesday", week.weekNumber, week.year)}</th>
          <th>Thursday {createDate("Thursday", week.weekNumber, week.year)}</th>
          <th>Friday {createDate("Friday", week.weekNumber, week.year)}</th>
        </thead>
        <tbody>
        <tr>
          <th></th>
          <td><Day key={week.monday.id} {...week.monday}/></td>
          <td><Day key={week.tuesday.id} {...week.tuesday}/></td>
          <td><Day key={week.wednesday.id} {...week.wednesday}/></td>
          <td><Day key={week.thursday.id} {...week.thursday}/></td>
          <td><Day key={week.friday.id} {...week.friday}/></td>
        </tr>
        </tbody>
      </table>
    </div>
  );
}

export default Week;
