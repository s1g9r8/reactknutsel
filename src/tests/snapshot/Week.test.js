
import React from 'react';
import ReactDOM from 'react-dom';
import Week from './../../components/Week.js';
import renderer from 'react-test-renderer';

it('should render a week', () => {
  const tree = renderer.create(<Week />).toJSON();
  expect(tree).toMatchSnapshot();
});
