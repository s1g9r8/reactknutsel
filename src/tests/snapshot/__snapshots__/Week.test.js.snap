// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a week 1`] = `
<div>
  <table>
    <caption>
      Week 
    </caption>
    <thead>
      <th />
      <th>
        Monday 
      </th>
      <th>
        Tuesday 
      </th>
      <th>
        Wednesday 
      </th>
      <th>
        Thursday 
      </th>
      <th>
        Friday 
      </th>
    </thead>
    <tbody>
      <tr>
        <th>
          project x
        </th>
        <td>
          <input
            name="monday"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </td>
        <td>
          <input
            name="tuesday"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </td>
        <td>
          <input
            name="wednesday"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </td>
        <td>
          <input
            name="thursday"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </td>
        <td>
          <input
            name="friday"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
